src/proto/generation.proto

```diff
@@ -345,6 +345,43 @@ message TransformResample {
   optional TransformMatrix prev_transform = 3; // 3x3 or 4x4 matrix, defaults to identity
   optional float depth_warp = 4;               // depth warp factor, defaults to 1.0
   optional bool export_mask = 5;               // return inpaint mask
+  optional bool do_prefill = 6;                // prefill border areas with values matching the colors around the area
+}
+
+message PointCloudRenderSettings {
+  float radius = 1;            // The radius (in NDC units) of each disk to be rasterized 
+  uint32 points_per_pixel = 2;  // Number of points to keep track of per pixel
+}
+
+message MeshRenderSettings {
+  float max_mesh_edge = 1;  // Mesh edges longer that this values are discarded
+}
+
+message RenderSettings {
+  oneof render_settings {
+    PointCloudRenderSettings pointcloud_settings = 1;
+    MeshRenderSettings mesh_settings = 2;
+  }
+}
+
+enum CameraType {
+  CAMERA_PERSPECTIVE = 0;   // Perspective camera
+  CAMERA_ORTHOGRAPHIC = 1;  // Orthographic camera. Well suited for isometric animations
+}
+
+message CameraSettings {
+  float near_plane = 1;    // Nearest plane depth of a rendered frustum
+  float far_plane = 2;     // Farthest plane depth of a rendered frustum
+  optional float fov = 3;  // Camera field of view (in angles). Must be set for CAMERA_PERSPECTIVE type.
+}
```

Comment:
Can TransformCameraPose just take another 4x4 TransformMatrix for the perspective transform like 3D resample? Have you found exposing near_plane and far_plane to be worthwhile?